tests/test_estimator.py

```diff
@@ -187,6 +187,30 @@ def kneighbors(self, X, custom_kwarg, **kwargs):
     )
 
 
+@parametrize_model_data()
+def test_kneighbors_nodata_outputs(model_data: ModelData):
```

Comment:
I have no idea if this is worth testing because it would be user error, but how about the possibility where `return_distances=False` and the user provides a tuple for `nodata_output`, e.g.:
```
nn = estimator.kneighbors(X_image, return_distance=False, nodata_output=(-32768, 255))
assert np.unique(unwrap_features(nn)) == [255]
```
This does raise an error, but perhaps not a totally intuitive message:
```
ValueError: Unsupported feature type: <class 'tuple'>
```
Your docstring is really clear that only in the case of `return_distance=True` should the user provide a tuple for `nodata_output`, so feel free to disregard.

Reply:
Good point! I think this error is easy enough to make that it's worth a special case. This now raises with `nodata_output must be a scalar when return_distance is False.`